lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -72,6 +72,10 @@
         pass
 
 
+class AnsibleParamikoConnectionFailure(AnsibleConnectionFailure):
```

Comment:
I dont see the need for a 'connection module specific' error class, the base connection failure works well enough and if we ever need the connection type, we should already have it in the caller.